targets/TARGET_STM/lp_ticker.c

```diff
@@ -203,6 +203,10 @@ void lp_ticker_set_interrupt(timestamp_t timestamp)
     __HAL_LPTIM_CLEAR_FLAG(&LptimHandle, LPTIM_FLAG_CMPM);
     __HAL_LPTIM_COMPARE_SET(&LptimHandle, timestamp);
 
+    while (__HAL_LPTIM_GET_FLAG(&LptimHandle, LPTIM_FLAG_CMPOK) == RESET) {
+    }
```

Comment:
Can you explain what is being done here ? Isn't there a risk to spend "a lot" of time in this while loop ?

Reply:
Yes, this would cause blocking behavior in a critical section which is problematic. Without this delay the call to `lp_ticker_clear_interrupt` doesn't actually stop a spurious ticker interrupt from firing. Is it possible to prevent a spurious interrupt here without blocking? If so could you share the sequence so I could incorporate it into this PR?

Alternatively, if a fix cannot be found, I could modify the test to ignore spurious interrupts when LPTICKER_DELAY_TICKS > 0. The LowPowerTickerWrapper layer filters out spurious interrupts so when enabled these will not propagate to higher layers. What are you thoughts on this @LMESTM?